Source/VFECore/Abilities/Ability.cs

```diff
@@ -91,7 +91,9 @@ public virtual bool IsEnabledForPawn(out string reason)
                 : current / (this.pawn.GetStatValue(statFactor.stat) * Math.Abs(statFactor.value));
         
         private float CalculateStatOffsetForPawn(float current, StatModifier statOffset) =>
-            current + this.pawn.GetStatValue(statOffset.stat) * statOffset.value;
+            new[]{ToStringStyle.PercentZero, ToStringStyle.PercentOne, ToStringStyle.PercentTwo}.Contains(statOffset.stat.toStringStyle) ? 
```

Comment:
I would like feedback on this. I'm not sure if this is the best way to check if a stat is expressed as a percentage relative to the base value.

Reply:
This seems unnecessarily confusing. Why can't you simply add the value of the stat to the current value?

Reply:
> This seems unnecessarily confusing. Why can't you simply add the value of the stat to the current value?

There are different kinds of stats - some are expressed as a percentage where we would only want to offset by the amount above/below the base level. Some are straight numbers, where we would want to use it as a true offset.

For example:
Psychic sensitivity - statOffset * (currentSensitivity - baseSensitivity)
Movespeed (c/s) - statOffset * moveSpeed

Reply:
There's one other thing I'm struggling with here - there are two "kinds" of offsets that could be desired. One expressed as a static value, and the other expressed as a percentage of the base value. To elaborate:

AbilityA has a range of 30.
I could desire to offset it by (5 * psychicSensitivity), which is how it currently works.
I could also desire to offset it by (5% of range * psychicSensitivity), which in this case would be (1.5 * psychicSensitivity)

I think the latter is more useful, but I'm not sure if this would be confusing behavior.

Reply:
> Psychic sensitivity - statOffset * (currentSensitivity - baseSensitivity)

Wouldn't this cause weird issues? Since if you are 150% psychic sensitivty the way that code would work is that it would add half the offset, which sounds not like what you want. I think it's a better bet to just add `offset * value`.

> There's one other thing I'm struggling with here - there are two "kinds" of offsets that could be desired. One expressed as a static value, and the other expressed as a percentage of the base value.

I think for anything more complex then static values we can simply say they have to use C# and override the corresponding calculator method.

Reply:
> > Psychic sensitivity - statOffset * (currentSensitivity - baseSensitivity)
> 
> Wouldn't this cause weird issues? Since if you are 150% psychic sensitivty the way that code would work is that it would add half the offset, which sounds not like what you want. I think it's a better bet to just add `offset * value`.
> 
> > There's one other thing I'm struggling with here - there are two "kinds" of offsets that could be desired. One expressed as a static value, and the other expressed as a percentage of the base value.
> 
> I think for anything more complex then static values we can simply say they have to use C# and override the corresponding calculator method.

That would mean a stat at 100% would still affect the result, where it's instead desired (at least in my case) to scale the offset relative to how far the stat is above/below the base value.

For example:
Ability has a range of 30 and I add a sensitivity offset of 10. The offset indicates that the range will increase by 10 per 100% sensitivity above base.

Sensitivity 100%: 30 (base range)
Sensitivity 150%: 30 + 5
Sensitivity 200%: 30 + 10
Sensitivity 50%: 30 - 5

This is similar to how consciousness affects manipulation.

Reply:
Ok I see what you mean. I think I agree with you. Instead of making an array and checking `Contains` inline, can you make a `private static HashSet<ToStringStyle>` and check against that?

Reply:
> Ok I see what you mean. I think I agree with you. Instead of making an array and checking `Contains` inline, can you make a `private static HashSet<ToStringStyle>` and check against that?

Sure, something to this effect? I moved the ternary statement to an if block to make the syntax clearer.
```c#
        private static readonly HashSet<ToStringStyle> percentageBasedStyles = new HashSet<ToStringStyle>
        {
            ToStringStyle.PercentZero,
            ToStringStyle.PercentOne,
            ToStringStyle.PercentTwo
        };
        
        private float CalculateStatOffsetForPawn(float current, StatModifier statOffset)
        {
            if (percentageBasedStyles.Contains(statOffset.stat.toStringStyle))
            {
                return current + (this.pawn.GetStatValue(statOffset.stat) - statOffset.stat.defaultBaseValue) *
                    statOffset.value;
            }

            return current + this.pawn.GetStatValue(statOffset.stat) * statOffset.value;
        }
```

Reply:
Yeah exactly, though I don't mind the ternary.

Reply:
Great. If I'm expecting modders to override this in harmony (for the percentage offset, for example) should the method be public?

Reply:
No, it's an internal method, so leave it private. People can use the name if they need to patch it.